src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -62,17 +59,20 @@ public void Dispose()
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
-            if (this._listenerStarted)
+            if (Interlocked.CompareExchange(ref this._listenerStatus, ListenerRegistering, ListenerNotRegistered) != ListenerNotRegistered)
             {
-                throw new InvalidOperationException("The listener has already been started.");
+                throw new InvalidOperationException("The listener has already been started or is starting.");
             }
 
             this.InitializeHost();
 
             try
             {
                 await this._host.RegisterObserverFactoryAsync(this);
-                this._listenerStarted = true;
+                if (Interlocked.CompareExchange(ref this._listenerStatus, ListenerRegistered, ListenerRegistering) != ListenerRegistering)
+                {
+                    throw new InvalidOperationException("The listener has already been started or was stopped during initialization.");
```

Comment:
You may not want to throw here -- just let it silently stay at "NotRegistered"? That would typically mean that Cancel() was called during StartAsync, in which case we probably just want to exit silently without touching the status. I think removing the throw would accomplish that. Does that seem right?